### PR TITLE
Integrate source-build patches into source

### LINKED ISSUE #160: ArPow followup: incorporate the .patch files added to the repo
The infra enabling ArPow (arcade-powered source-build) involves some `.patch` files checked into this repo at `eng/source-build-patches/*.patch` that fix up the repo to work for source-build. These patch files are prone to conflicting with dev changes, and may block some PRs until the PR author fixes up the patch file to match. The changes in these `.patch` files should be incorporated into the main branch of this repo to remove this possibility.

Some background on source-build patches, for anyone who isn't familiar with previous pushes for patch incorporation:

A patch is essentially just a commit that has been extracted from Git into a `.patch` file that can be applied on demand. The effort to build .NET from source involves creating patches because repos make changes that are incompatible with source-build and need to be fixed up after the original released source code has already been finalized. When the original repo gets PRs over time for servicing, the PR changes sometimes conflict with the source-build patches, just like a merge conflict. The patch files need to be fixed up when this happens, which is a significant maintenance problem for the source-build team.

Several times, the source-build team has pushed for "patch incorporation". This means to merge the commit represented in the `.patch` file into the original repo's official branch. Doing so prevents patch merge conflicts, because there's no longer a patch to merge against. However, patches inevitably pile up again when getting subsequent servicing releases to work in source-build.

ArPow lets us end this maintenance-heavy process. By running source-build inside CI, patch merge conflicts will immediately block PR validation, so fixup can be handled in place, not solely by the source-build team. Running source-build in CI also means creating new patches won't be necessary except in exceptional circumstances.

This issue tracks using `git am eng/source-build-patches/*.patch` and submitting a PR to incorporate the patches, or making changes to the repo that accomplish the same goals as the patches.

For <https://github.com/dotnet/source-build/blob/master/Documentation/planning/arcade-powered-source-build/implementation-plan.md>

### PR DESCRIPTION
**Pull request by MichaelSimons**
Fixes #160

In addition to removing the source-build patches, I also added the missing SourceBuild attribute to the arcade Versions.Details.xml file.  This singles source-build to utilize the source-build intermediate package for arcade.

### CODE REVIEW

**Comment by MichaelSimons** [comment]
cc @crummel, please review at your convenience.

**Review by crummel — APPROVED** [review]

**Review by Sanan07 — APPROVED** [review]

**Comment by MichaelSimons** [comment]
Thanks for the review @Sanan07.  I don't have permission to merge.  Can you please merge at your convenience?

**Comment by Sanan07** [comment]
Sure, I will merge

File: eng/SourceBuild.props [hunk 1]
```diff
@@ -5,19 +5,4 @@
     <SourceBuildManagedOnly>true</SourceBuildManagedOnly>
   </PropertyGroup>
 
-  <Target Name="ApplySourceBuildPatchFiles"
```
Comment (Haplois) [inline eng/SourceBuild.props:8]: Keeping this part for future patches could be helpful, since it would do nothing if there is not patches present.

Comment (MichaelSimons) [reply eng/SourceBuild.props:8]: We don't intend to have/allow patches going forward.  We want the product to support source-build natively.  Adding source build legs to the CI will help ensure the product is source-buildable when changes are introduced.

**Pull request merged**